src/Config/Parser.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WPML\PB\Config;
+
+use WPML\FP\Lst;
+use WPML\FP\Obj;
+
+class Parser {
+
+	/** @var string $configRoot */
+	private $configRoot;
+
+	/** @var string $defaultConditionKey */
+	private $defaultConditionKey;
+
+	public function __construct( $configRoot, $defaultConditionKey ) {
+		$this->configRoot          = $configRoot;
+		$this->defaultConditionKey = $defaultConditionKey;
+	}
+
+	/**
+	 * Receives a raw config array (from XML) and convert it into
+	 * a page builder configuration array.
+	 *
+	 * @see WPML_Elementor_Translatable_Nodes::get_nodes_to_translate()
+	 *
+	 * @param array $allConfig
+	 *
+	 * @return array
+	 */
+	public function extract( array $allConfig ) {
+		$pbConfig = [];
+		$allWidgets = Obj::path( [ 'wpml-config', $this->configRoot, 'widget' ], $allConfig );
+
+		if ( $allWidgets ) {
+			foreach ( $allWidgets as $widget ) {
+				$widgetName = Obj::path( ['attr', 'name'], $widget );
+
+				$pbConfig[ $widgetName ] = [
+					'conditions' => $this->parseConditions( $widget, $widgetName ),
+					'fields'     => $this->parseFields( $widget ),
+				];
+
+				$fieldsInItem = Obj::prop( 'fields-in-item', $widget );
+
+				if ( $fieldsInItem ) {
+					$itemOf                                    = Obj::path( [ 'attr', 'items_of' ], $fieldsInItem );
+					$pbConfig[ $widgetName ]['fields_in_item'] = [ $itemOf => $this->parseFieldsInItem( $fieldsInItem ) ];
+				}
+
+				$integrationClasses = $this->parseIntegrationClasses( $widget );
+
+				if ( $integrationClasses ) {
+					$pbConfig[ $widgetName ]['integration-class'] = $integrationClasses;
+				}
+			}
+		}
+
+		return $pbConfig;
+	}
+
+	/**
+	 * @param array  $widget
+	 * @param string $widgetName
+	 *
+	 * @return array
+	 */
+	private function parseConditions( array $widget, $widgetName ) {
+		$rawConditions = Obj::path( ['conditions', 'condition'], $widget );
+
+		if ( $rawConditions ) {
+			$parsedConditions = [];
+
+			foreach ( $this->normalize( $rawConditions ) as $condition ) {
+				$parsedConditions[ Obj::pathOr( $this->defaultConditionKey, ['attr', 'key'], $condition ) ] = $condition['value'];
+			}
+		} else {
+			$parsedConditions = [ $this->defaultConditionKey => $widgetName ];
+		}
+
+		return $parsedConditions;
```

Comment:
@strategio You could try something like this. You'll need to make `normalize` public for this.

```php
$makePair = function( $condition ) { 
    return [Obj::pathOr( $this->defaultConditionKey, ['attr', 'key'], $condition ), $condition['value'] ] 
};
return Maybe::fromNullable( Obj::path( ['conditions', 'condition'], $widget ) )
   ->map( [ $this, 'normalize ] )
   ->map( Fns::map( $makePair ) )
   ->map( Lst::fromPairs() )
   ->getOrElse(  [ $this->defaultConditionKey => $widgetName ] );
```

Reply:
Excellent @brucepearson , done!